api/types/src/transaction.rs

```diff
@@ -523,6 +555,16 @@ pub struct BlockMetadataTransaction {
     pub timestamp: U64,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct SystemTransaction {
+    #[serde(flatten)]
+    #[oai(flatten)]
+    pub info: TransactionInfo,
+    pub events: Vec<Event>,
+    pub timestamp: U64,
+    pub txn_serialized: Vec<u8>,
```

Comment:
Should this be an enum?

Reply:
probably. But let's also expect this enum will have new variant every ~3 months. Will that hurt?

Reply:
I don't see any problem here.